### ISSUE TITLE
Apply `directives_ordering` to `part`

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/linter#4945]
**Describe the issue**
There is the `directives_ordering`. It does not apply to `part`, only `import` and `export`.

**To Reproduce**
- Create a simple project.
- Ensure `directives_ordering` is being used:

```yaml
linter:
  rules:
    directives_ordering: true
```

- Create 2 new files `a.dart` and `b.dart` inside `lib`.
- Make them `part` of `main.dart`:

```dart
part 'b.dart';
part 'a.dart';
```

**Expected behavior**
To trigger `directives_ordering` lint.

**Additional context**
If you disagree with adding this rule to `directives_ordering`, then maybe we could create a new linter to that.

**Comment by bwilkerson** [comment]
It does seem reasonable to me that the lint should check `part` directives.

**Comment by pq** [comment]
Chatting w/ @jakemac53, sorting parts is not something we can safely do in a world with augmentations because then order matters.

Because of that, I don't think we want to go ahead with this after all.

**Comment by FMorschel** [comment]
I'm not familiar with augmentations yet, but can't the analyzer see if this is being used to apply the suggestion only when it is not? Or would it be weird to do so?

**Comment by jakemac53** [comment]
Fwiw, it is possible to determine when it is safe to do. But, implementing that might not be worth while (or spending the compute on it). It is safe if any of the following are true:

- If none of the parts have augmenting declarations
- The augmenting declarations only introduce new members (and don't alter existing ones)
- Any existing member which is augmented, is only augmented once.

But, it doesn't feel write for this lint to be computing all of that, when today it can operate on just a single parsed AST.

**Comment by FMorschel** [comment]
Thats understandable. Could you please link here some docs for augmentations so I can see what you are talking about? And if there is nothing else you'd like to discuss here we can close this issue.
